repos/public-user/src/Module/SiteSettingExtension.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+
+namespace PublicUser\Module;
+
+
+use Omeka\Form\SiteSettingsForm;
+use Zend\Form\Element\Checkbox;
+use Zend\Form\Element\Select;
+use Zend\Form\Element\Text;
+use Zend\Form\Fieldset;
+
+trait SiteSettingExtension
+{
+
+    public function createCheckbox(SiteSettingsForm $form, string $id, string $label, bool $default = true)
+    {
+        return (new Checkbox($id))
+            ->setOptions([
+                'label' => $label,
+            ])
+            ->setValue($form->getSiteSettings()->get($id, $default));
+    }
+
+    public function extendSiteSettings(SiteSettingsForm $form)
+    {
+        $form->add(
+            (new Fieldset('public-user'))
```

Comment:
Doesn't this produce an object that looks like:

```
[
    'public_user' => [
         'public-user-enable-registration' => ...,
    ]
]
```

Seems like you could omit the `public-user-` prefix in the fields themselves.

Reply:
Unfortunately not, we could namespace them, like the configuration forms: https://github.com/digirati-co-uk/madoc-platform/blob/master/repos/iiif-storage/src/Admin/ConfigurationForm.php 

Same result in the end in terms of `name` props on the form, but much easier to scan and read.

Reply:
Ah, these aren't configuration forms. Got it.

Reply:
Not the admin config forms, they are fieldsets in each of the individual sites. It would be nice if they could added to in a similar way with a similar API. Not sure exactly how yet though